File: src/amata1219/xeflection/kotlin/MinecraftPackage.kt
package amata1219.xeflection.kotlin

import org.bukkit.Bukkit

sealed class MinecraftPackage(packageName: String) {
    val path: String

    init {
        val version: String = Bukkit.getServer().javaClass.`package`.name.replaceFirst(
                Regex(".*(\\d+_\\d+_R\\d+).*"),
                "$1"
        )
        path = "$packageName.v$version"
    }

    override fun toString(): String = path
}
class NetMinecraftServer : MinecraftPackage("net.minecraft.server")
class OrgBukkitCraftbukkit: MinecraftPackage("org.bukkit.craftbukkit")

File: src/amata1219/xeflection/scala/MinecraftPackage.scala
package amata1219.xeflection.scala

import org.bukkit.Bukkit

sealed class MinecraftPackage(packageName: String) {
  val path: String = {
    val version = Bukkit.getServer.getClass.getPackage.getName.replaceFirst(
      ".*(\\d+_\\d+_R\\d+).*",
      "$1"
    )
    s"$packageName.v$version"
  }

  override def toString: String = path
}
case object NetMinecraftServer extends MinecraftPackage("net.minecraft.server")
case object OrgBukkitCraftbukkit extends MinecraftPackage("org.bukkit.craftbukkit")